### PR TITLE
Bump min_ruby_version to 2.4

### PR DESCRIPTION
**Pull request by dogenpunk**
This pull request is leading up to a fix for hanami/hanami#921

Add TargetRubyVersion to rubocop config to silence warnings about
mismatch between rubocop config and gemspec.
Switch =~ usage to match? per rubocop in files.remove_block

It appears that jruby 9.2.0.0 is available on travis per: https://travis-ci.org/rubocop-hq/rubocop/jobs/398743238

### CODE REVIEW

**Comment by dogenpunk** [comment]
It looks like JRuby 9.2 *is* available on Travis. I'll resubmit the pull requests for the other projects in the next day or so.

**Comment by jodosha** [comment]
@dogenpunk Thanks for this PR. The TravisCI page still doesn't report JRuby 9.2 http://rubies.travis-ci.org/ 😞 

---

I've changed the base branch because this is a breaking change. I don't think that t

Quoting the original [ticket](https://github.com/hanami/hanami/issues/921)

> Changes MUST go to unstable branches of each repository

**Comment by dogenpunk** [comment]
Right on, thanks for taking the time to clean up my mistakes! 👍 I'll make sure to submit the other PRs to `unstable`

**Comment by AlfonsoUceda** [comment]
@dogenpunk FYI https://github.com/hanami/devtools/pull/2 has been merge

File: .rubocop.yml [hunk 1]
```diff
@@ -1,5 +1,7 @@
 # Please keep AllCops, Bundler, Style, Metrics groups and then order cops
 # alphabetically
+AllCops:
+  TargetRubyVersion: 2.4
 inherit_from:
   - https://raw.githubusercontent.com/hanami/devtools/master/.rubocop.yml
```
Comment (AlfonsoUceda) [inline .rubocop.yml]: @jodosha should we update this `inherit_from` key with `.rubocop_unstable.yml`?

Comment (jodosha) [reply .rubocop.yml]: @AlfonsoUceda Yes, and `TargetRubyVersion` should be updated in `.rubocop_unstable.yml`.

Comment (AlfonsoUceda) [reply .rubocop.yml]: :+1:

**Comment by jodosha** [comment]
We can't merge yet this change because as now, `hanami-model` and `hanami-view` aren't compatible with JRuby 9.2 yet.

**Comment by jodosha** [comment]
Closed in favor of https://github.com/hanami/utils/pull/288. @dogenpunk Thanks for your help. 👍

**Pull request closed without merging**